internal/common/update/patch_payload.go

```diff
@@ -177,3 +207,11 @@ func PatchPayload[T any](state, plan *T) (*T, error) {
 	}
 	return patchReq, json.Unmarshal(reqJSON, patchReq)
 }
+
+func IsEmpty[T any](last *T) bool {
```

Comment:
nit: it looks like here we're checking if all struct attributes have zero values, empty can sound more like an empty collection (slice, map, etc.), don't know if it's better IsZero or similar

Reply:
Renamed to `IsZeroValues` WDYT?

Reply:
it's fine, thanks